### ISSUE TITLE
[Bug] Training RFDETR, DINOv2 and VLM silently overwrites previous training runs

### ISSUE THREAD

**Issue by gboeer** [Libre-YOLO/libreyolo#833]
**What happened**

Calling .train() on LibreRFDETR, LibreDINOv2, or the LibreVLM LoRA SFT tier through the Python API (not the CLI) writes into the same output directory every time instead of creating an incremented run folder, **silently overwriting the previous run's checkpoints**. Every other family (YOLO9, D-FINE, DEIM, YOLO-NAS, PP-YOLOE, RT-DETR/v2, YOLOX, YOLOv7, SegFormer, etc.) correctly increments (exp, exp2, exp3, ...).

**Reproduce**

```py
from libreyolo import LibreYOLO

model = LibreYOLO("LibreRFDETRn.pt")
model.train(data="data.yaml", epochs=1)   # writes to runs/train
model.train(data="data.yaml", epochs=1)   # overwrites runs/train
```
Same pattern reproduces with LibreDINOv2.train() and the LibreVLM LoRA SFT trainer.

**Root cause**

RFDETR.train() and DINOv2.train() popped exist_ok from kwargs with a hardcoded local default of True, instead of deferring to their own config class's default (False, same as every other family). This disabled the shared BaseTrainer._get_save_dir() increment-on-exists logic whenever a caller didn't explicitly pass exist_ok=False. VLMTrainConfig.exist_ok had the identical wrong default (True) in its own independent save-dir resolution.

The bug is masked when training through the CLI, because _build_rfdetr_train_kwargs() in cli/config.py pre-increments and pre-creates the run directory itself before calling train(), it only bites direct Python API use.

**Comment by EHxuban11** [comment]
Hello @gboeer , 

Thank you for your issue. Nice catch, very sharp as always : )
